src/plot.c

```diff
@@ -129,12 +129,15 @@ plot_plot(struct plot *plot)
 		return 0;
 	}
 
+
 	/* Determine the max and min of the array*/
-	struct plot_bounds bounds = plot_data_get_bounds(plot->datasets, plot->data);
+	if (!plot->fixed_bounds) {
+		plot->bounds = plot_data_get_bounds(plot->datasets, plot->data);
+	}
 
-	plot_make_labels(plot, &bounds);
+	plot_make_labels(plot, &plot->bounds);
 
-	plot_normalize_data(plot, &bounds);
+	plot_normalize_data(plot, &plot->bounds);
```

Comment:
If you are making `bounds` a member of `plot`, you should modify `plot_make_labels` and `plot_normalize_data` to only take `plot` as an argument to avoid redundancy

Reply:
Acknowledged, done in the next patchset.